gtars/src/io/gtok.rs

```diff
@@ -0,0 +1,190 @@
+use std::fs::File;
+use std::fs::OpenOptions;
+use std::io::{BufReader, BufWriter, Read, Write};
+
+use anyhow::{Context, Result};
+
+use super::consts::{GTOK_HEADER, GTOK_U16_FLAG, GTOK_U32_FLAG};
+
+///
+/// Writes a vector of tokens to a file in the `.gtok` format.
+/// # Arguments
+/// - filename: the file to save the tokens to
+/// - tokens: tokens to save
+///
+pub fn write_tokens_to_gtok(filename: &str, tokens: &[u32]) -> Result<()> {
+    // make sure the path exists
+    let path = std::path::Path::new(filename);
+
+    if let Some(parent) = path.parent() {
+        std::fs::create_dir_all(parent)?;
+    } else {
+        anyhow::bail!("Failed to create parent directories for gtok file!")
+    }
+
+    let file = File::create(filename).with_context(|| "Failed to create gtok file!")?;
+    let mut writer = BufWriter::new(file);
+
+    // write the header
+    writer
+        .write_all(GTOK_HEADER)
+        .with_context(|| "Failed to write GTOK header to file!")?;
+
+    // determine size of tokens
+    let is_small = tokens.iter().all(|&x| x <= u16::MAX as u32);
+    let flag = if is_small {
+        GTOK_U16_FLAG
+    } else {
+        GTOK_U32_FLAG
+    };
+    writer
+        .write_all(&flag.to_le_bytes())
+        .with_context(|| "Failed to write GTOK size flag to file!")?;
+
+    for &token in tokens {
+        if is_small {
+            writer
+                .write_all(&(token as u16).to_le_bytes())
+                .with_context(|| "Failed to write bytes to file!")?;
+            continue;
+        }
+        writer
+            .write_all(&token.to_le_bytes())
+            .with_context(|| "Failed to write bytes to file!")?;
+    }
```

Comment:
suggestion
```suggestion
        if is_small {
            for &token in tokens {
            writer
                .write_all(&(token as u16).to_le_bytes())
                .with_context(|| "Failed to write bytes to file!")?;
            }
        } else {
            for &token in tokens {
        writer
            .write_all(&token.to_le_bytes())
            .with_context(|| "Failed to write bytes to file!")?;
            }
      }
```

I'm pretty sure the indent level is off, but the point is moving the if outside the loop.